include/ginkgo/core/solver/solver_base.hpp

```diff
@@ -449,6 +449,22 @@ class SolverBase {
             typeid(*vec), size, size[1]);
     }
 
+    template <typename LinOpType>
+    LinOpType* create_workspace_op_with_type_of(int vector_id,
+                                                const LinOpType* vec,
+                                                dim<2> global_size,
+                                                dim<2> local_size) const
+    {
+        return workspace_.template create_or_get_op<LinOpType>(
+            vector_id,
+            [&] {
+                return LinOpType::create_with_type_of(
```

Comment:
Is it feasible to test this behavior?

Reply:
In what sense? I added tests for the `create_with_type_of` for `Vector`. I will also add a test for the two size version of `Dense`.

Reply:
specifically the workspace functionality, not the Vector functionality

Reply:
First of all, these are kind of implicitly tested through all the concrete solver tests. Also, if we would test this, I think it makes sense to test the other workspace functions. But, I think that should be tackled in a separate PR.